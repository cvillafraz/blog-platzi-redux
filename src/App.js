import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom'
import Layout from './components/Layout'
import Users from './screens/Users'
import Posts from './screens/Posts'
import store from './config/store'
import { Provider } from 'react-redux'

const usersUrl = "https://jsonplaceholder.typicode.com/users"
const postsUrl = "https://jsonplaceholder.typicode.com/posts" 

    
const App = () => { 

	return <Provider store={store}>
		<BrowserRouter>
			<Layout>
				<Switch>
					<Route exact path='/' render={()=><Users url={usersUrl} />} />
					<Route path='/publicaciones/:key' render={()=><Posts postsUrl={postsUrl}
					usersUrl={usersUrl} />} />
				</Switch>
			</Layout>
		</BrowserRouter>
	</Provider>
}

export default App;