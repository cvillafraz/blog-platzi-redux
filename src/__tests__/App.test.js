import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import { create } from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
describe("App component", () => {
  test("Matches app snapshot", () => {
    const button = create(<App />);
    expect(button.toJSON()).toMatchSnapshot();
  });
});
