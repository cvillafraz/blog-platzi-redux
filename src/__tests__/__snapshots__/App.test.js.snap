// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Matches app snapshot 1`] = `
Array [
  <nav
    id="menu"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Usuarios
    </a>
    <a
      href="/tareas"
      onClick={[Function]}
    >
      Tareas
    </a>
  </nav>,
  <div
    className="margen"
  >
    <h2>
      Usuarios
    </h2>
    <table
      className="tabla"
    >
      <thead>
        <tr>
          <th>
            Nombre
          </th>
          <th>
            Correo
          </th>
          <th>
            Enlace
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>,
]
`;
